src/cluster_legacy.c

```diff
@@ -186,6 +191,54 @@ dictType clusterSdsToListType = {
     NULL                /* allow to expand */
 };
 
+typedef struct {
+    enum { ITER_DICT, ITER_LIST } type;
+    union {
+        dictIterator di;
+        listIter li;
+    };
+} ClusterNodeIterator;
+
+static void clusterNodeIterInitAllNodes(ClusterNodeIterator *iter) {
+    iter->type = ITER_DICT;
+    dictInitSafeIterator(&iter->di, server.cluster->nodes);
+}
+
+static void clusterNodeIterInitMyShard(ClusterNodeIterator *iter) {
+    list *nodes = clusterGetNodesInMyShard(server.cluster->myself);
+    serverAssert(nodes != NULL);
+    iter->type = ITER_LIST;
+    listRewind(nodes, &iter->li);
+}
+
+static clusterNode *clusterNodeIterNext(ClusterNodeIterator *iter) {
+    switch (iter->type) {
+    case ITER_DICT: {
+        /* Get the next entry in the dictionary */
+        dictEntry *de = dictNext(&iter->di);
+        /* Return the value associated with the entry, or NULL if no more entries */
+        return de ? dictGetVal(de) : NULL;
+    }
+    case ITER_LIST: {
+        /* Get the next node in the list */
+        listNode *ln = listNext(&iter->li);
+        /* Return the value associated with the node, or NULL if no more nodes */
+        return ln ? listNodeValue(ln) : NULL;
+    }
+    /* This line is unreachable but added to avoid compiler warnings */
+    default: {
+        serverPanic("bad type");
+        return NULL;
+    }
+    }
```

Comment:
I know its a bit annoying, but Clang wont let me do that. It would fail with that change.
```
[Clang Format Check/clang-format-check] ⭐ Run Main Check for formatting changes
[Clang Format Check/clang-format-check]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/3.sh] user= workdir=
| ERROR: Code is not formatted correctly. Here is the diff:
| diff --git a/src/cluster_legacy.c b/src/cluster_legacy.c
| index 02c9bb9ff..6feef5618 100644
| --- a/src/cluster_legacy.c
| +++ b/src/cluster_legacy.c
| @@ -213,26 +213,23 @@ static void clusterNodeIterInitMyShard(ClusterNodeIterator *iter) {
|  
|  static clusterNode *clusterNodeIterNext(ClusterNodeIterator *iter) {
|      switch (iter->type) {
| -    case ITER_DICT:
| -        {
| +    case ITER_DICT: {
|          /* Get the next entry in the dictionary */
|          dictEntry *de = dictNext(&iter->di);
|          /* Return the value associated with the entry, or NULL if no more entries */
|          return de ? dictGetVal(de) : NULL;
| -        }
| -    case ITER_LIST: 
| -        {
| +    }
| +    case ITER_LIST: {
|          /* Get the next node in the list */
|          listNode *ln = listNext(&iter->li);
|          /* Return the value associated with the node, or NULL if no more nodes */
|          return ln ? listNodeValue(ln) : NULL;
| -        }
| +    }
|      /* This line is unreachable but added to avoid compiler warnings */
| -    default:
| -        {
| +    default: {
|          serverPanic("bad type");
|          return NULL;
| -        }
| +    }
|      }
|  }
```

Reply:
Ok, let's follow clang-format then.